### PR TITLE
[ADD] get_model2table: method to map nonstandard table names

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Extracted from https://github.com/OCA/OpenUpgrade/blob/16.0/openupgrade_scripts/scripts/base/16.0.1.3/pre-migration.py#L65C9-L65C64, in order to all versions get the benefit.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Not sure if this may vary across versions.

**Comment by MiquelRForgeFlow** [comment]
> Not sure if this may vary across versions.

 I have checked from v8 to v16 and this is the only change:
Selection_1476

So, I think we are being safe enough. If in the future any of these models/tables changes, then we will adapt this new method accordingly.

File: openupgradelib/openupgrade_merge_records.py [hunk 1]
```diff
@@ -217,10 +217,11 @@ def _change_reference_refs_sql(
         try:
             model = env[row[0]]
         except KeyError:
-            continue
-        if not model._auto:  # Discard SQL views
-            continue
-        table = model._table
+            table = get_model2table(row[0])
+        else:
+            if not model._auto:  # Discard SQL views
```
Comment (pedrobaeza) [inline openupgradelib/openupgrade_merge_records.py:222]: This is not evaluated in the proper place.

File: openupgradelib/openupgrade_merge_records.py [hunk 2]
```diff
@@ -758,8 +759,10 @@ def _change_generic(
         try:
             model = env[model_to_replace].with_context(active_test=False)
         except KeyError:
-            continue
-        if (model._table, res_id_column) in exclude_columns:
+            table = get_model2table(model_to_replace)
+        else:
+            table = model._table
```
Comment (pedrobaeza) [inline openupgradelib/openupgrade_merge_records.py:764]: Don't do this in `else`, but next to the `model = env[model_to_replace].with_context(active_test=False)`. The same for all occurrences of this type.

File: openupgradelib/openupgrade_merge_records.py [hunk 3]
```diff
@@ -861,7 +864,10 @@ def _delete_records_sql(
     env, model_name, record_ids, target_record_id, model_table=None
 ):
     if not model_table:
-        model_table = env[model_name]._table
+        try:
```
Comment (pedrobaeza) [inline openupgradelib/openupgrade_merge_records.py:867]: Not sure about all of this new `try`. If the model is not available, another bigger problem may arise, so I'm thinking that better to detect it sooner than later.

File: openupgradelib/openupgrade_merge_records.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply openupgradelib/openupgrade_merge_records.py:222]: What do you mean? The `else` part is executed if no exception is triggered. Thus, the code is fine.

File: openupgradelib/openupgrade_merge_records.py (hunk 3 above)
Comment (MiquelRForgeFlow) [reply openupgradelib/openupgrade_merge_records.py:867]: This is `_delete_records_sql()`, which used only when method='sql'. My point is we should be able to use the merge records with method='sql' in tables that are obsolete or not currently in the env in the moment the merge is called.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**